graylog2-server/src/main/java/org/graylog2/users/UserAndTeamsConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.users;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import org.graylog.autovalue.WithBeanGetter;
+
+@JsonAutoDetect
+@AutoValue
+@WithBeanGetter
+public abstract class UserAndTeamsConfig {
+    @JsonProperty
+    public abstract boolean sharingWithEveryone();
+
+    @JsonProperty
+    public abstract boolean sharingWithUsers();
+
+    @JsonCreator
+    public static UserAndTeamsConfig create(
+            @JsonProperty("allow_sharing_with_everyone") boolean sharingWithEveryone,
```

Comment:
These property names are not the same ones that are used when serializing.
I think this only works because of JsonAutoDetect.
But better give everything a consistent naming.

Reply:
Weird that this even works. I'd rather it just failed. Good catch.